src/ansys/platform/instancemanagement/instance.py

```diff
@@ -31,54 +30,84 @@
 logger = logging.getLogger(__name__)
 
 
-@dataclass
 class Instance(contextlib.AbstractContextManager):
     """Provides a remote instance of a product.
 
     This class is a context manager and can be used with the ``with`` statement to
     automatically stop the remote instance when the tasks are finished.
     """
 
-    definition_name: str
-    """Name of the definition that created this instance."""
+    _stub: ProductInstanceManagerStub
 
-    name: str
-    """Name of the instance.
+    _definition_name: str
+    _name: str
+    _ready: bool
+    _status_message: str
+    _services: Mapping[str, Service]
 
-    This name is chosen by the server and always start with ``"instances/"``."""
+    @property
+    def definition_name(self) -> str:
+        """Name of the definition that created this instance."""
+        return self._definition_name
 
-    ready: bool
-    """Whether the instance is ready.
+    @property
+    def name(self) -> str:
+        """Name of the instance.
 
-    If ``True``, the ``services`` field contains the list of entry points
-    exposed by the instance.
-
-    If ``False``, the ``status_message`` field contains a human-readable
-    reason."""
+        This name is chosen by the server and always start with ``"instances/"``.
+        """
+        return self._name
 
-    status_message: str
-    """Status of the instance.
+    @property
+    def ready(self) -> bool:
+        """Whether the instance is ready.
 
-    Human-readable message describing the status of the instance.
-    This field is always filled when the instance is not ready."""
+        If ``True``, the ``services`` field contains the list of entry points
```

Comment:
suggestion: replace `field` by `property` in the doc strings for the `ready`, `status_message`, and `services` properties.